docs/migration-recipe.md

```diff
@@ -0,0 +1,106 @@
+Migrating to new host:
+----------------------
+
+### Prepare for migration:  
+
+1.  Ensure that salt-master, loadbalancer, and host in question are up, and that their health check for the relevant service is failing in haproxy after the host is fully up
+
+To view haproxy status: 
+
+-vagrant up the salt-master, loadbalancer, and host in question (`vagrant up`)
+
+-`vagrant ssh-config salt-master loadbalancer >> vagrant-ssh Ssh -L 4646:127:0.0.1:464 -F vagrant-ssh loadbalancer`
+
+-open loacalhost:4646/haproxy?stats in browser
+
+2.  Edit pillar data for roles.sls to include both old and new hostnames (ex. hostname*)
+
+### Migrate the host:
+
+#### Update Salt Master with latest config including prep from above
+
+1.  ssh into the salt-master server `ssh salt-master`
+
+2.  Navigate to srv/psf-salt `cd srv/psf-salt`
+
+3.  Run `git pull`
+
+4.  Run `sudo salt-call state.highstate`
+
+#### Ensure new configuration doesn't impact host being migrated
+
+1.  ssh into the host in question `ssh hostname`
+
+2.  Navigate to srv/psf-salt `cd srv/psf-salt`
+
+3.  Run `git pull`
+
+4.  Run `sudo salt-call state.highstate`
+
+#### Create new host:
+
+1.  Start a new droplet  in digital ocean, and check resources being used on old host to see if we are over or under spending on resources 
+
+2.  Create a new droplet with new version of ubuntu, appropriate resources,  and name it according to hostname + 2004
+
+#### Provision new host for migration:
+
+1.  Add Salt repositories for our current target version (add the apt-repo and install salt-minion package)
+
+-  `RUN wget --quiet -O /usr/share/keyrings/salt-archive-keyring.gpg https://repo.saltproject.io/py3/ubuntu/20.04/$(dpkg --print-architecture)/3004/salt-archive-keyring.gpg`
+
+-  `RUN echo "deb [signed-by=/usr/share/keyrings/salt-archive-keyring.gpg arch=$(dpkg --print-architecture)] https://repo.saltproject.io/py3/ubuntu/20.04/$(dpkg --print-architecture)/3004 focal main" > /etc/apt/sources.list.d/salt.list`
+
+2.  Configure the salt-minion. On new host, run the command
+
+- `RUN apt-get update -y && apt-get install -y --no-install-recommends salt-minion`
+
+- On the original host, look through `etc/salt/minion.d*` to set up salt-minion configuration files to match on new host 
+
+3.  Run `sudo salt-call state.highstate`
```

Comment:
Unfortunately, I don't think that high state will succeed at this point.

We're missing a few steps.

Namely,

- Restart the salt-minion service on the new host to pickup the configuration and register with the salt-master
- Accept the key for the new host on the salt-master